plugins/woocommerce/changelog/fix-cli-and-endpoints

```diff
@@ -0,0 +1,6 @@
+Significance: minor
+Type: fix
+
+- Adding header validation to prevent this error:
+`PHP Warning:  Undefined array key "X-WP-Total" in /app/wordpress/wp-content/plugins/woocommerce/includes/cli/class-wc-cli-rest-command.php on line 239`
+- Adding `X-WP-Total` and `X-WP-TotalPages` header on `tax_class` and `payment_gateway` endpoints.
```

Comment:
Changelogs are compiled into the final readme.txt so ideally they would be short. I'm offering up a suggestion but feel free to adapt as you see fit.

```suggestion
Add validation to prevent PHP warning in CLI when using the 'count' format.
```

Given this PR also takes care of adding support for some endpoints, I'd suggest you add another changelog file for the other changes. You can use the same `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` tool to create a new one.